packages/server/lib/StudioLifecycleManager.ts

```diff
@@ -0,0 +1,132 @@
+import type { StudioManager } from './cloud/studio'
+import { ProtocolManager } from './cloud/protocol'
+import { getAndInitializeStudioManager } from './cloud/api/studio/get_and_initialize_studio_manager'
+import Debug from 'debug'
+import type { CloudDataSource } from '@packages/data-context/src/sources'
+import type { Cfg } from './project-base'
+import _ from 'lodash'
+import type { DataContext } from '@packages/data-context'
+import api from './cloud/api'
+
+const debug = Debug('cypress:server:studio-lifecycle-manager')
+const routes = require('./cloud/routes')
+
+export class StudioLifecycleManager {
+  private studioManagerPromise: Promise<StudioManager | null> | null = null
+  private studioReady = false
+  private studioManager: StudioManager | null = null
+  private listeners: ((studioManager: StudioManager) => void)[] = []
+  /**
+   * Initialize the studio manager and possibly set up protocol.
+   * Also registers this instance in the data context.
+   * @param projectId The project ID
+   * @param cloudDataSource The cloud data source
+   * @param cfg The project configuration
+   * @param debugData Debug data for the configuration
+   * @param ctx Data context to register this instance with
+   */
+  initializeStudioManager ({
+    projectId,
+    cloudDataSource,
+    cfg,
+    debugData,
+    ctx,
+  }: {
+    projectId?: string
+    cloudDataSource: CloudDataSource
+    cfg: Cfg
+    debugData: any
+    ctx: DataContext
+  }): void {
+    debug('Initializing studio manager')
+
+    const studioManagerPromise = getAndInitializeStudioManager({
+      projectId,
+      cloudDataSource,
+    }).then(async (studioManager) => {
+      if (studioManager.status === 'ENABLED') {
+        debug('Cloud studio is enabled - setting up protocol')
+        const protocolManager = new ProtocolManager()
+        const protocolUrl = routes.apiRoutes.captureProtocolCurrent()
+        const script = await api.getCaptureProtocolScript(protocolUrl)
+
+        await protocolManager.prepareProtocol(script, {
+          runId: 'studio',
+          projectId: cfg.projectId,
+          testingType: cfg.testingType,
+          cloudApi: {
+            url: routes.apiUrl,
+            retryWithBackoff: api.retryWithBackoff,
+            requestPromise: api.rp,
+          },
+          projectConfig: _.pick(cfg, ['devServerPublicPathRoute', 'port', 'proxyUrl', 'namespace']),
+          mountVersion: api.runnerCapabilities.protocolMountVersion,
+          debugData,
+          mode: 'studio',
+        })
+
+        studioManager.protocolManager = protocolManager
+        studioManager.isProtocolEnabled = true
+      } else {
+        debug('Cloud studio is not enabled - skipping protocol setup')
+      }
+
+      debug('Studio is ready')
+      this.studioReady = true
+      this.studioManager = studioManager
+      this.callRegisteredListeners()
+
+      return studioManager
+    }).catch((err) => {
+      debug('Error during studio manager setup: %o', err)
```

Comment:
Let's notify the ErrorReportManager here, and let's clean up the listeners here. In the future we may want to notify the listeners that there was an error, but let's not worry about that for now.